### ISSUE TITLE
Updating Joi Dependency

### ISSUE THREAD

**Issue by chrisforrette** [holidayextras/jsonapi-server#170]
Hello,

I noticed the version of Joi in this project is 6.10.1 and it's up to version 9.0. Is this something that's reasonably easy to upgrade? Thanks!

**Comment by pmcnr-hx** [comment]
Joi > 6 doesn't support Node.js 0.10 (which `jsonapi-server` still does), therefore we have to stay at `joi-6.x` at the moment. Do you have any compelling reason for requiring more modern recent versions of Joi?

**Comment by tdreyno** [comment]
I'd really suggest getting support up to Node 4.4, which is their Long Term Support release. I think most of the community has moved in that direction (hence joi dropping old 0.10 support).

Anyways, joi keeps adding nice helper methods that we'd hoping to use in our schemas.

**Comment by theninj4** [comment]
You're all right that we should move to the most recent version. The main thing holding me back is the main reason why this project exists and why I've poured countless hours in to it is that the project currently using this module in production is running on 0.10.x, so if I bump `Joi` I won't be able to consume my own module ;)

Last time I checked (many months ago) a lot of commercial usage of Node was pre-v4, although I'm sure that graph is rapidly swinging around. We're well on the road to node v4 and v6, when our commercial project makes the jump rest assured this project will be right behind it.  

Amusingly we've had to move and rebuild our CI and production environments to move to v4 due to newer versions of glibc. It's taken longer than I'd have expected. We're probably about 5 days off now.

**Comment by theninj4** [comment]
https://semaphoreci.com/blog/2015/12/15/nodejs-version-usage-in-commercial-projects-2015-edition.html
